rtengine/ipwavelet.cc

```diff
@@ -2216,67 +2223,57 @@ void ImProcFunctions::ip_wavelet(LabImage * lab, LabImage * dst, int kall, const
-                        float amean = 0.5f / (tempmean);
-                    
+                        constexpr float insigma = 0.666f; //SD
+                        const float logmax = log(tempmax); //log Max
+                        const float rapX = (tempmean + sig * tempsig) / tempmax; //rapport between sD / max
+                        const float inx = log(insigma);
+                        const float iny = log(rapX);
+                        const float rap = inx / iny; //koef
+                        const float asig = 0.166f / (tempsig * sig);
+                        const float bsig = 0.5f - asig * tempmean;
+                        const float amean = 1.f / tempmean;
 
 #ifdef _OPENMP
         #pragma omp parallel for schedule(dynamic, Wlvl_L * 16) num_threads(wavNestedLevels) if (wavNestedLevels>1)
 #endif
 
                         for (int i = 0; i < Wlvl_L * Hlvl_L; i++) {
                             float absciss;
-                            float tempwav = 0.f;
-                            tempwav = 0.7f * WavCoeffs_L[dir][i] + 0.3f * WavCoeffs_L2[dir][i];
+                            const float tempwav = std::fabs(0.7f * WavCoeffs_L[dir][i] + 0.3f * WavCoeffs_L2[dir][i]);
 
-                            if (std::fabs(tempwav) >= (tempmean + sig * tempsig)) { //for max
-                                float valcour = xlogf(std::fabs(tempwav));
-                                float valc = valcour - logmax;
-                                float vald = valc * rap;
+                            if (tempwav >= tempmean + sig * tempsig) { //for max
+                                const float vald = (xlogf(tempwav) - logmax) * rap;
                                 absciss = xexpf(vald);
-                            } else if (std::fabs(tempwav) >= tempmean) {
-                                absciss = asig * std::fabs(tempwav) + bsig;
+                            } else if (tempwav >= tempmean) {
+                                absciss = asig * tempwav + bsig;
                             } else {
-                                absciss = amean * std::fabs(tempwav);
-                                float abs = pow(2.f * absciss, (1.f / SQR(sig)));
-                                absciss = 0.5f * abs;
-                            }
-                            float kc = wavguid.getVal(absciss) -1.f;
-
-                            if(kc < 0) {
-                                kc = -SQR(kc);//approximation to simulate sliders denoise
+                                absciss = amean * tempwav;
+                                if (sig == 2.f) { // for sig = 2.f we can use a faster calculation because the exponent in this case is 0.25
+                                    absciss = 0.5f * std::sqrt(std::sqrt(absciss));
+                                } else { // original code for variable sig
+                                    absciss = 0.5f * pow_F(absciss, 1.f / k);
```

Comment:
I am not be following the transformation correctly, but isn't there a factor two missing inside the calls to `pow_F` and `sqrt` that was in the old code:
```c++
absciss = amean * std::fabs(tempwav);
float abs = pow(2.f * absciss, (1.f / SQR(sig)));
absciss = 0.5f * abs;
```

Reply:
That factor of two went into an outside of loop calculation

Before it was
```
float amean = 0.5f / (tempmean);
```
Now it is
```
const float amean = 1.f / (tempmean);
```

Reply:
Thanks! No objections then.